DDD Workshop.ipynb

```diff
@@ -0,0 +1,547 @@
+    "    \n",
+    "    key = \"{}_{}\".format(airport, time)\n",
+    "    if key not in data_by_airport_time_raw.keys():\n",
+    "        data_by_airport_time_raw[key] = []\n",
+    "        data_by_airport_time_clean[key] = []\n",
+    "    \n",
+    "    raw_occupancy = float(raw_line[2])\n",
+    "    clean_occupancy = int(clean_line[2]) / int(clean_line[1])\n",
+    "    \n",
+    "    data_by_airport_time_raw[key].append(raw_occupancy)\n",
+    "    data_by_airport_time_clean[key].append(clean_occupancy)\n",
+    "    \n",
+    "data_file.close()\n",
+    "cleaned_data_file.close()\n",
+    "\n",
+    "\n",
+    "# processing data into raw bar chart\n",
+    "rate_by_airport_time_raw = {}\n",
+    "for key, data_arr in data_by_airport_time_raw.items():\n",
+    "    occupancy_average = sum(data_arr) / len(data_arr)\n",
+    "    rate_by_airport_time_raw[key] = occupancy_average\n",
+    "\n",
+    "plt.figure(figsize=(17, 3))\n",
+    "names = rate_by_airport_time_raw.keys()\n",
+    "values = rate_by_airport_time_raw.values()\n",
+    "plt.ylabel('occupancy count')\n",
+    "plt.suptitle('Raw Airport Occupancy Rate')\n",
+    "plt.bar(names, values)\n",
+    "plt.show()\n",
+    "\n",
+    "\n",
+    "# processing data into cleaned bar chart\n",
+    "rate_by_airport_time_clean = {}\n",
+    "for key, data_arr in data_by_airport_time_clean.items():\n",
+    "    occupancy_average = sum(data_arr) / len(data_arr)\n",
+    "    rate_by_airport_time_clean[key] = occupancy_average\n",
+    "\n",
+    "\n",
+    "plt.figure(figsize=(17, 3))\n",
+    "names = rate_by_airport_time_clean.keys()\n",
+    "values = rate_by_airport_time_clean.values()\n",
+    "plt.bar(names, values, align='center')\n",
+    "plt.ylabel('occupancy percent')\n",
+    "plt.suptitle('Fixed Airport Occupancy Percentage')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We see the issue with the original data set very clearly here, and we see that the conclusion from part 2 is still correct (JFK Redeye has the highest percentage), but we also see how small the percentage difference is.\n",
+    "\n",
+    "\n",
+    "There are many ways to communicate the conclusions that you have gathered, and there are many ways that those conclusions can be presented. We want to make sure that in our effort to convince, we're not stepping over the line of manipulating the other person. There are some heinous examples of misleading visualizations, some of the worst posted online and easy to search up in [aggreator lists](https://analythical.com/blog/examples-of-awful-data-visualization). It is worth taking some time to see the ways these can be misleading too, as the common mistakes can creep into your visualizations too if you're not careful. I've found [this guide from venngage](https://venngage.com/blog/misleading-graphs/) quite useful in familiarizing myself with the do not's of data viz.\n",
```

Comment:
Condense: It is worth taking time to see the ways these can be misleading too, as the common mistakes can creep into your visualizations if you're not careful.